### PR TITLE
DEPR: deprecate (Sparse)Series.from_array

### LINKED ISSUE #18213: Deprecate Series.from_array ?
I just noticed we have a `Series.from_array` method (I was not conscientiously aware of that :-))

Is there any reason you would want to use that instead of the normal `pd.Series(..)` constructor ?

### PR DESCRIPTION
**Pull request by jorisvandenbossche**
Closes #18213. 

It's a bit annoying that I have to keep it as internal method. On the other hand this might signal that `from_array` actually has some use case ..

### CODE REVIEW

File: pandas/core/series.py [hunk 1]
```diff
@@ -273,6 +273,25 @@ def __init__(self, data=None, index=None, dtype=None, name=None,
...
+
+        """
+        warnings.warn("'from_array' is deprecated and will be removed in a "
+                      "future version. Please use the pd.Series(..) "
+                      "constructor instead.", FutureWarning, stacklevel=2)
+        cls._from_array(arr, index=index, name=name, dtype=dtype, copy=copy,
+                       fastpath=fastpath)
+
+    @classmethod
+    def _from_array(cls, arr, index=None, name=None, dtype=None, copy=False,
```
Comment (jreback) [inline pandas/core/series.py]: its prob easier just to elminate this and deal with SparseSeries directly in the Series constructor. Do we have an issue for this?

Comment (jreback) [reply pandas/core/series.py]: annoyhing we'd actually have to do this in ``__new__`` (but not that big of a deal)

Comment (jorisvandenbossche) [reply pandas/core/series.py]: Yes, that would be cleaner. But it is also an API change that `Series(..)` would no longer always return a Series but depending on the input also SparseSeries. 
So the question is if we actually want this. I can open an issue about it though.

**Comment by jreback** [comment]
thanks!

**Comment by gfyoung** [comment]
@jorisvandenbossche @jreback : This commit introduced the first (and only) instance of `pytest.warns` instead of our in-house `tm.assert_produces_warning`.  Do we have a preference for which should be used now that we require `pytest >= 3.1` ?

**Comment by jreback** [comment]
@gfyoung let's just use ``assert_produces_warning``. and alway pls add a lint rule to fail on``pytest.warns``. We could of course change to use ``pytest.warns`` (maybe) but that's a different issue.

**Comment by gfyoung** [comment]
@jreback : I figured as such.  Thanks for confirming.

**Pull request merged**